mono/metadata/sgen-mono.c

```diff
@@ -1738,7 +1738,7 @@ report_registered_roots_by_type (int root_type)
 	RootRecord *root;
 	report.count = 0;
 	SGEN_HASH_TABLE_FOREACH (&sgen_roots_hash [root_type], void **, start_root, RootRecord *, root) {
-		SGEN_LOG (6, "Profiler root scan %p-%p (desc: %p)", start_root, root->end_root, (void*)root->root_desc);
+		SGEN_LOG (6, "Profiler root scan %p-%p (desc: %p)", start_root, root->end_root, (void*)(intptr_t)root->root_desc);
```

Comment:
`intptr_t` or `uintptr_t`?

Reply:
`SgenDescriptor` is an unsigned `target_mword`, so I guess `uintptr_t` is slightly better...

Reply:
Probably it should mword aka host_mword.

Reply:
Unless we build them at compile-time.

Reply:
Which wouldn't be surprising actually.